cpp/src/arrow/compute/kernels/scalar_string.cc

```diff
@@ -3421,41 +3415,45 @@ const FunctionDoc utf8_length_doc("Compute UTF8 string lengths",
 
 void AddStrptime(FunctionRegistry* registry) {
   auto func = std::make_shared<ScalarFunction>("strptime", Arity::Unary(), &strptime_doc);
```

Comment:
Hmm, strptime explicitly works on string inputs - I don't think it makes sense to try to parse timestamps from a binary string? (Unless you have odd data with binary delimiters?)

Reply:
I agree that `strptime` should only support string inputs. It only registers kernels for `StringTypes`.